bbcode-src/bbscript/bbscript1/functions.js

```diff
@@ -0,0 +1,416 @@
+    const [leftResult, rightResult] = evaluate(options, lhs, rhs);
+    return leftResult < rightResult;
+  },
+};
+/**
+ * is less than or equal to
+ * @type {bbscriptFunc}
+ */
+export const leq = {
+  params,
+  func: (options, lhs, rhs) => {
+    const [leftResult, rightResult] = evaluate(options, lhs, rhs);
+    return leftResult >= rightResult;
+  },
+};
+export const random = {
+  params: [{ types: [bbscriptParamTypes.Int] }, { types: [bbscriptParamTypes.Int] }],
+  func: (options, min, max) => {
+    return Math.floor(Math.random() * (max - min + 1) + min);
+  },
+};
+export const print = {
+  params: [{ types: [bbscriptParamTypes.String, bbscriptParamTypes.Identifier] }],
+  func: (options, args) => {
+    ConsoleLogger.log(getStringVal(args, options));
+  },
+};
+/**
+ * Set variable
+ * @type {bbscriptFunc}
+ */
+export const set = {
+  params: [
+    { types: [bbscriptParamTypes.Identifier] },
+    { types: [bbscriptParamTypes.Int, bbscriptParamTypes.String, bbscriptParamTypes.Function] },
+  ],
+  func: (options, id, value) => {
+    if (isAstNode(value)) {
+      value = options.processor.exec(value, options);
+    } else {
+      value = getStringVal(value, options);
+    }
+    const callerId = options.callerId || "";
+    if (!(callerId in options.data)) {
+      options.data[callerId] = {};
+    }
+    options.data[callerId][id] = value;
+  },
+};
+export const add = {
+  params: [
+    {
+      types: [
+        bbscriptParamTypes.Int,
+        bbscriptParamTypes.String,
+        bbscriptParamTypes.Identifier,
+        bbscriptParamTypes.Function,
+      ],
+    },
+    {
+      types: [
+        bbscriptParamTypes.Int,
+        bbscriptParamTypes.String,
+        bbscriptParamTypes.Identifier,
+        bbscriptParamTypes.Function,
+      ],
+    },
+  ],
+  func: (options, a, b) => {
+    if (isAstNode(a)) {
+      a = options.processor.exec(a, options);
+    } else {
+      a = getStringVal(a, options);
+    }
+    if (isAstNode(b)) {
+      b = options.processor.exec(b, options);
+    } else {
+      b = getStringVal(b, options);
+    }
+    return a + b;
+  },
+};
+export const dec = {
+  params: [
+    { types: [bbscriptParamTypes.Identifier] },
+    { types: [bbscriptParamTypes.Int], default: 1 },
+  ],
+  func: (options, id, amount = 1) => {
+    try {
+      options.data[id] -= amount;
+    } catch (e) {
+      ConsoleLogger.warn(`${id} is not a number`, e);
+    }
+  },
+};
+
+export const inc = {
+  params: [
+    { types: [bbscriptParamTypes.Identifier] },
+    { types: [bbscriptParamTypes.Int], default: 1 },
+  ],
+  func: (options, id, amount = 1) => {
+    try {
+      options.data[id] += amount;
+    } catch (e) {
+      ConsoleLogger.warn(`${id} is not a number`, e);
+    }
+  },
+};
+
+/**
+ * Map between bbscript function name and logic.
+ * keys must be all lowercase. Parser will apply lowercase to user input,
+ * so is case-insensitive.
+ * @type {bbscriptFuncMap}
+ */
+export const bbscriptFunctions = {
```

Comment:
Mentioned in the discord, just pulling over here so we keep comments in one place:
This PR includes adding functionality back in for BBScript. We're should go ahead and update the wiki so we have some documentation on what to test and the extent of the functionality to make sure BBScript doesn't open the app up for XSS exploitation.